### PR TITLE
Adds 2 new RecipeParser functions for Anaconda Linter

### PR DESCRIPTION
**Pull request by schuylermartin45**
- Adds a `find_value()` function to find values in the parse tree
- Adds a `add_selector()` function to allow callers to modify recipe selectors
- Adds unit tests

Both of these new functions are being added to automatically fix the `no_git_on_windows` Anaconda Linter rule.

### CODE REVIEW

File: percy/render/recipe_parser.py [hunk 1]
```diff
@@ -1310,6 +1341,55 @@ def get_selector_paths(self, selector: str) -> list[str]:
...
+
+        if node is None:
+            raise KeyError(f"Path not found: {path!r}")
+        selector_re = re.compile(r"\[.*\]")
+        if not selector_re.match(selector):
+            raise ValueError(f"Invalid selector provided: {selector}")
+
+        # Helper function that extracts the outer set of []'s in a selector
+        def _extract_selector(s: str) -> str:
+            return s.replace("[", "", 1)[::-1].replace("]", "", 1)[::-1]
```
Comment (schuylermartin45) [inline percy/render/recipe_parser.py:1366]: This is arguably not the best way to handle extracting the `[]` from a selector, but it gets this PR over the line.

File: percy/render/recipe_parser.py [hunk 2]
```diff
@@ -1310,6 +1341,55 @@ def get_selector_paths(self, selector: str) -> list[str]:
...
+        elif old_selector_found:
+            logic_op = "and" if mode == SelectorConflictMode.AND else "or"
+            old_selector = _extract_selector(old_selector_found.group())
+            new_selector = _extract_selector(selector)
+            comment = f"# [{old_selector} {logic_op} {new_selector}]"
+        # If the comment is not a selector, put the selector first, then append
+        # the comment.
+        else:
+            # Strip the existing comment of it's leading `#` symbol
+            comment = f"# {selector} {node.comment.replace('#', '', 1).strip()}"
```
Comment (JeanChristopheMorinPerso) [inline percy/render/recipe_parser.py:1382]: I'm not sure this is supported by conda-build. If I remember correctly, last time I tried conda-build failed when a trailing comment was found.

Comment (schuylermartin45) [reply percy/render/recipe_parser.py:1382]: I guess that makes things easier.
I suppose I should probably wipe the comment then? Does that sound like a reasonable approach to default to "replace" in that instance?

Comment (schuylermartin45) [reply percy/render/recipe_parser.py]: If that's the case, the next commit simplifies the logic to do just that.

Comment (JeanChristopheMorinPerso) [reply percy/render/recipe_parser.py]: I just tested and it looks like `# [osx] asasd` and `# [osx] # asasd` work. SO I guess I was wrong.

**Review by JeanChristopheMorinPerso — APPROVED** [review]
LGTM, though I didn't take a deep look. I see tests and they seem good to me.

File: percy/render/recipe_parser.py (hunk 2 above)
Comment (schuylermartin45) [reply percy/render/recipe_parser.py]: In that case, I will redact that commit and merge. I'd rather have the logic in place. Thanks for checking that though! It was a good call out to see if it was supported.

**Pull request merged**